### DISCUSSION TITLE
A suggestion about introducing new level when calling "ptypy.core.Ptycho"

### DISCUSSION THREAD

**Discussion by Amirafsharm** [ptycho/ptypy#465] [Ideas]
I often would like to try different parameters of an engine, but each time I change an engine parameter, e.g., number of iterations or learning rate, I need to wait againg for ptypy to load the data.
Could we expect a modification in future so that data preparation is separated from engine creation?

**Comment by daurer** [comment]
This is now possible with the introduction of a new feature that copies the current object/probe state which can then be restored later. See #506
